### PR TITLE
Added getSourceIds to the controller

### PR DESCRIPTION
**Pull request by mariusvn**
Added getSourceIds to the controller with the android (tested) and ios (no ability to test here sorry). Not implemented on the web as getLayerIds in not implemented too

I made the controller method returning a list of strings (`Future<List<String>>`) to have an explicit `String` list unlike `getLayerIds` who have a `Future<List<dynamic>>`

### CODE REVIEW

File: ios/Classes/MapboxMapController.swift [hunk 1]
```diff
@@ -765,6 +765,17 @@ class MapboxMapController: NSObject, FlutterPlatformView, MGLMapViewDelegate, Ma
             var reply = [String: NSObject]()
             reply["layers"] = layerIds as NSObject
             result(reply)
+
+        case "style#getSrouceIds":
```
Comment (m0nac0) [inline ios/Classes/MapboxMapController.swift]: I think there's a typo here?

File: lib/src/controller.dart [hunk 2]
```diff
@@ -1270,6 +1270,12 @@ class MaplibreMapController extends ChangeNotifier {
     return _mapboxGlPlatform.getLayerIds();
   }
 
+  Future<List<String>> getSourceIds() async {
```
Comment (m0nac0) [inline lib/src/controller.dart:1276]: Could you please add a short doc comment here, including the fact that it's not implemented on web, yet?

Comment (mariusvn) [reply lib/src/controller.dart:1276]: yup, no problems

File: ios/Classes/MapboxMapController.swift (hunk 1 above)
Comment (mariusvn) [reply ios/Classes/MapboxMapController.swift]: yes, let me fix it

**Comment by mariusvn** [comment]
the requested changes are done like you asked @m0nac0

**Comment by m0nac0** [comment]
Sorry for noticing only now: could you add a button somewhere in the example app using this new function? That way we could manually test it (now and in the future)

**Comment by mariusvn** [comment]
Yes, I was hesitant to do this when adding the feature so i'll do it later today

Edit: tomorrow

**Comment by mariusvn** [comment]
I added a page with get layer ids and get source ids

**Comment by m0nac0** [comment]
Great thanks! I enabled auto-merge, if you could again update with the latest changes from main it should merge automatically. (Btw, I was able to confirm it works on iOS)

**Pull request merged**